api/v1alpha1/selfnoderemediationconfig_webhook_test.go

```diff
@@ -119,6 +119,34 @@ var _ = Describe("SelfNodeRemediationConfig Validation", func() {
 
 	})
 
+	Describe("deleting SelfNodeRemediationConfig CR", func() {
+		var conf *SelfNodeRemediationConfig
+		BeforeEach(func() {
+			conf = createDefaultSelfNodeRemediationConfigCR()
+		})
+
+		When("SelfNodeRemediationConfig CR is default", func() {
+			BeforeEach(func() {
+				conf.Name = "self-node-remediation-config"
+				//Mock deployment namespace
+				_ = os.Setenv("DEPLOYMENT_NAMESPACE", conf.Namespace)
+			})
+			It("should have warning", func() {
+				war, err := conf.ValidateDelete()
+				Expect(err).To(Succeed())
+				Expect(war[0]).To(ContainSubstring("The default configuration is deleted, Self Node Remediation is now disabled"))
+			})
+		})
+
+		When("SelfNodeRemediationConfig CR is non default", func() {
```

Comment:
In theory you are correct, however the create part belongs to different functionality so IMO no harm in testing that part as well in the delete module (it might become more relevant in case create functionality is modified).

Nothing I feel strongly about, so if you feel the confusion it creates override the value let me know and I'll remove the test.

Reply:
I wonder if ignoring the deletion of non-default SNRC _only_ is a feature.
We are not intentionally allowing that, because we don't allow creating it in the first place.
It is kind of a side effect of the implementation we chose, while the feature we want is preventing the creation of the 2nd and the deletion of the last one, but this would require to _count_ the configurations.

Reply:
NP, test removed.